dialogy/plugins/text/address_parser/__init__.py

```diff
@@ -0,0 +1,212 @@
+from typing import Any, Dict, List, Optional
+import copy
+
+from dialogy.base.plugin import Plugin, Input, Output
+from dialogy.types import Intent
+from dialogy.types.slots import Slot
+
+from maps import (
+    get_address_from_mapmyindia_geocode,
+    get_matching_address_from_gmaps_autocomplete,
+)
+
+
+class GoogleMapsAPIPlugin(Plugin):
+    def __init__(
+        self,
+        dest=None,
+        access: Optional[Plugin] = None,
+        address_capturing_intents: List[str] = None,
+        pincode_capturing_intents: List[str] = None,
+        country_code: str = None,
+        language: str = None,
+        debug: bool = False,
+    ) -> None:
+        """A Plugin to hit the Google Maps API to fill the slot with the returned address"""
+        super().__init__(
+            debug=debug,
+        )
+        self.address_capturing_intents = address_capturing_intents
+        self.pincode_capturing_intents = pincode_capturing_intents
+        self.country_code = country_code
+        self.language = language
+
+    def utility(self, input_: Input, output: Output) -> Any:
+        intents: List[Intent]
+        ner_input: List[str]
+
+        intents = output.intents
+        context = input_
+
+        pincode = None
+
+        if context.history:
+            for intent in context.history[0]["intents"]:
+                if intent["name"] in self.pincode_capturing_intents:
+                    for slot in intent["slots"]:
+                        if slot["name"] == "pincode":
+                            pincode = str(slot["values"][0]["value"]).replace(" ", "")
+
+        first_intent, *rest = intents
+        slots: List[Dict[str, Any]] = first_intent.slots
+
+        if (not intents) or (not isinstance(intents, list)):
+            return
+
+        if not slots:
+            return
+
+        if first_intent.name in self.address_capturing_intents:
+            ner_input = [
```

Comment:
You can add an attribute called `best_transcript` within `Input`. and set its value like so:

```python
    def __attrs_post_init__(self) -> None:
        try:
            object.__setattr__(self, "transcripts", normalize(self.utterances))
            object.__setattr__(self, "best_transcript", fn(self.utterances)) <--- add just this.
        except TypeError:
            ...

```